File: opensource/DataBind_Lamda/app/src/main/java/com/tao/databind_lamda/view/MainActivity.java
package com.tao.databind_lamda.view;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.tao.databind_lamda.R;
import com.tao.databind_lamda.databinding.ActivityMainBinding;
import com.tao.databind_lamda.viewmodel.MainViewModel;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding mBinding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        mBinding.setViewModel(new MainViewModel(MainActivity.this));
    }
}

File: Demo/KotlinDemo/app/src/main/java/com/tao/kotlindemo/MainActivity.kt
package com.tao.kotlindemo

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*



class MainActivity : AppCompatActivity() {
    var intents = Intent("com.tao.foregroud.service")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_onclick01.requestFocus()

//        btn_onclick01.setOnClickListener {
//            view -> ToastUtils.showToast(view.context,"Hello Kotlin !!")
//        }

        btn_onclick01.setOnClickListener {
            view -> run{ ToastUtils.showToast(view.context,"hello Kotlin")} }


        btn_start_service.setOnClickListener { view -> run {

            startService(intents)
            ToastUtils.showToast(view.context,"启动service服务！！")
        } }

        btn_stop_service.setOnClickListener { view -> run{
            stopService(intents)
        } }
    }

    private fun startMyService(){

    }

    override fun onStop() {
        super.onStop()
    }
}
